### PR TITLE
Room-games: dont rename players to their guest accounts

### PR DESCRIPTION
**Pull request by sparkychildcharlie**
- prevents users from being unable to rejoin games if they ``/logout`` (the server will rename you back to your guest account if you use /logout, thus barring re-entry or re-joining to certain IP limiting room-games)

### CODE REVIEW

**Comment by sparkychildcharlie** [comment]
(Apply the same change to UNO's editted rename scheme.)

File: room-game.js [hunk 1]
```diff
@@ -160,6 +160,7 @@ class RoomGame {
 			return;
 		}
 		if (!(oldUserid in this.players)) return;
+		if (!user.named) return; // allow users that have logged out to join back on the name they last used.
```
Comment (Zarel) [inline room-game.js]: This should be treated the same as the `!this.allowRenames` case.

**Comment by Zarel** [comment]
The problem here is that we do want to track people's renames through `/fr` etc...

**Comment by Zarel** [comment]
Maybe `/fr` should just provide an exception...

**Comment by sparkychildcharlie** [comment]
hm, ill do that tomorrow.   I was wondering, isn't forcerename used by global %s as a method to kick timer stallers as i remember being told they had no other tools at their disdispo?

**Comment by Zarel** [comment]
ugh, it really shouldn't be

**Comment by Zarel** [comment]
they're not supposed to deal with timerstallers at all

**Comment by CrestfallPS** [comment]
No one does that anymore, it did used to be done before you (Zarel) clarified timer stuff in that thread. I haven't seen any driver/mod do this since.

**Comment by sparkychildcharlie** [comment]
Ah I guess what i knew was from a long time ago :P 

Anyways, in addition to the above (not sticking users onto their guest accounts), this will:
- make force-renames NOT auto forfeit a battle
- allow Guests that are in a battle (from ``/forcerename``) to select a name
- track users onto their guest account after they get force-renamed

**Comment by Zarel** [comment]
Hm, that's probably the right approach, but let me think about it a bit.

**Comment by Zarel** [comment]
Yeah, this is non-ideal but I don't know of a better way to achieve this.

**Pull request merged**